portal/src/graphql/adminapi/AuditLogScreen.tsx

```diff
@@ -37,13 +43,37 @@ function CommandBarDropdownWrapper(props: ICommandBarItemProps) {
   return <CommandBarDropdown {...dropdownProps} />;
 }
 
+function RefreshButton(props: ICommandBarItemProps) {
+  const tooltipStyle: Partial<ITooltipHostStyles> = {
+    root: { display: "inline-block" },
+  };
+  const tooltipId = useId("refreshTooltip");
+  const tooltipCalloutProps = { gapSpace: 0 };
+  const tooltipcontent =
+    "Refresh. Updated: " + formatDatetime(props.locale, getCurrentDate());
+  return (
+    <TooltipHost
+      content={tooltipcontent}
+      styles={tooltipStyle}
+      id={tooltipId}
+      calloutProps={tooltipCalloutProps}
+      directionalHint={DirectionalHint.bottomCenter}
+    >
+      {/* @ts-expect-error */}
+      <CommandBarButton {...props} />
+    </TooltipHost>
+  );
+}
+
 const AuditLogScreen: React.FC = function AuditLogScreen() {
   const [offset, setOffset] = useState(0);
   const [selectedKey, setSelectedKey] = useState("ALL");
   const [dateRangeDialogHidden, setDateRangeDialogHidden] = useState(true);
   const [sortDirection, setSortDirection] = useState<SortDirection | undefined>(
     SortDirection.Desc
   );
+  const [currentTimeStamp, setCurrentTimeStamp] = useState(getCurrentDate());
+  const [isCustomDateRange, setIsCustomDateRange] = useState(false);
```

Comment:
Do not introduce derivable state. It is a custom date range with we have `rangeFrom`, otherwise it is not.